### PR TITLE
Setupscript

### PR DESCRIPTION
**Pull request by 402-matiaskj**
added setup scripts
symlinks files, creates a few folders
added nvim install script
installs a hardcoded version of nvim into ~/bin

### CODE REVIEW

**Comment by dr-thca** [comment]
Terrible pr, do not approve

**Comment by 402-matiaskj** [comment]
> Terrible pr, do not approve

Please do elaborate why this PR is «terrible» as you say. I wonder if your mom knows you are online spreading hate and bile, I’m sure she is thrilled to have spawned such a lovely creature. I was having a good day (i just flashed a 9c, boulder for the first time) until i read your acidic comment. Now my day is ruined. I was going to show my son how to ride the bike but now instead i will have to spend my energies recovering from this brutal attack (my son is crying, he was looking forward to learning how to ride the bike (a big milestone for a child in my country)). Congratulations you just destroyed the day (perhaps more) of a child. This is why we think before we speak, but I suspect that wisdom is lost on you. I hope you have a nice day (I’m a nice person like that) and i hope you call your mom and apologize for being such a dissapointment.

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		echo "deleted, symlinking from dotfiles repo"
+		cd ~; ln -s $SCRIPT_DIR/.zshrc .zshrc
+	# symlink if no exists
+	else
+		echo "zshrc does not exist, symlinking"
+		cd ~; ln -s $SCRIPT_DIR/.zshrc .zshrc
+	fi 
+else
+	echo "zsh already symlinked to dotfiles repo, skipping"
+fi
```
Comment (dr-thca) [inline setup.sh:45]:
```suggestion
ln -sf $SCRIPT_DIR/.zshrc ~/.zshrc
```

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+# create local-zshrc for all our local config
+if ! [ -f ~/.local-zshrc.sh ]; then
+	echo "creating local-zshrc"
+	touch ~/.local-zshrc.sh
+else
+	echo "local-zshrc already exists, skipping"
+fi
```
Comment (dr-thca) [inline setup.sh:53]:
```suggestion
touch ~/.local-zshrc.sh
```
touch er non-destructive

File: setup.sh [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		echo "deleted, symlinking from dotfiles repo"
+		cd ~; ln -s $SCRIPT_DIR/.tmux.conf .tmux.conf
+	# symlink if no exists
+	else
+		echo "tmux conf does not exist, symlinking"
+		cd ~; ln -s $SCRIPT_DIR/.tmux.conf .tmux.conf
+	fi 
+else
+	echo "tmux conf already symlinked to dotfiles repo, skipping"
+fi
```
Comment (dr-thca) [inline setup.sh:70]:
```suggestion
ln -fs $SCRIPT_DIR/.tmux.conf ~/.tmux.conf
```

File: setup.sh [hunk 4]
```diff
@@ -0,0 +1,91 @@
+#! /bin/bash
+
```
Comment (dr-thca) [inline setup.sh:2]:
```
set -euxo pipefail
```
https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425
Så scriptet feiler på evt error.

File: setup.sh [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+then 
+	echo "zsh not installed, system incompatible with config"
+	exit
+fi
+# check if shell is zsh, if it is not promt user and quit, 
+if [[ $(getent passwd | grep $USER | awk -F ":" '{print $7}' | grep zsh -c) == 0  ]]
+then
+	echo "shell not set to zsh, system incompatible with config"
+	exit
+fi
```
Comment (dr-thca) [inline setup.sh:21]:
```suggestion
if [[ cat $SHELL | grep zsh -c) == 0  ]]
```

File: setup.sh [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+		cd ~/.config; ln -s $SCRIPT_DIR/nvim nvim
+		
+	else
+		echo "nvim config does not exist, symlinking"
+		cd ~/.config/
+		cd ~/.config; ln -s $SCRIPT_DIR/nvim nvim
+	fi
+else
+	echo "nvim conf already symlinked to dotfiles repo, skipping"
+fi
```
Comment (dr-thca) [inline setup.sh:90]: force symlink

File: install-nvim.sh [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+install_nvim() {
+	# download and extract
+	mkdir -p ~/downloads
+	cd -p ~/downloads
+	rm ~/downloads/nvim-linux64.tar.gz
+	wget $nvim_tarball	
+	tar -xzf ~/downloads/nvim-linux64.tar.gz
+	
+	#symlink to bin
+	mkdir ~/bin
```
Comment (dr-thca) [inline install-nvim.sh:16]:
```
	mkdir -p ~/bin
```

File: install-nvim.sh [hunk 8]
```diff
@@ -0,0 +1,31 @@
+#! /bin/bash
+
```
Comment (dr-thca) [inline install-nvim.sh:2]:
```suggestion
set -exuo pipefail
```

**Review by dr-thca — COMMENTED** [review]
Also your commit messages are non-descriptive. Please rebase and reword the messages.

**Pull request merged**